### ISSUE TITLE
Generate syntax highlighting files

### ISSUE THREAD

**Issue by vanillajonathan** [francisrstokes/arcsecond#103]
After you create a parser then you have write syntax highlighting files to syntax highlight the language your parser parses. When doing changes to the parser you manually have to update these files.

It would be good if you could generate such syntax highlight files for:
- [Ace](https://github.com/ajaxorg/ace) JavaScript editor.
- [Lezer](https://lezer.codemirror.net/) for the [CodeMirror](https://codemirror.net/) JavaScript editor.
- [Monaco](https://microsoft.github.io/monaco-editor/), Microsoft's JavaScript editor used by VS Code.
- [GtkSourceView](https://gitlab.gnome.org/GNOME/gtksourceview/), the GTK source code editor component used by GNOME Text Editor, GNOME Builder and others.
- [KDE Syntax Highlighting Engine](https://invent.kde.org/frameworks/syntax-highlighting) used by Kate, KDevelop and others. ([API](https://api.kde.org/frameworks/syntax-highlighting/html/index.html))
- [highlight.js](https://highlightjs.org/), syntax highlighting for the web.
- [Sublime Text](https://www.sublimetext.com/docs/syntax.html), text editor
- [TextMate](https://macromates.com/manual/en/language_grammars), text editor

**Comment by francisrstokes** [comment]
Hey @vanillajonathan - I can see the problem this would cause. I'm not sure there is a way to solve this generically in the library, as arcsecond doesn't tie to your parser down to any particular structure.

Did you already have an idea about how this might be tackled?

**Comment by vanillajonathan** [comment]
No, I don't have any idea how this would be tackled, it was just a wild idea. I am unaware of how arcsecond works. Please feel free to close this issue if you deem it nonviable.

**Comment by odama626** [comment]
I just had a similar thought, it would be really cool if there was a supplemental package that took the arc second parser and created a language server from it.

Its what vscode / neovim / helix / maybe Monaco uses?

**Comment by odama626** [comment]
In the same way there could be supplemental packages for other editors something like @arcsecondjs/manaco etc
